br/pkg/lightning/backend/local/duplicate_test.go

```diff
@@ -70,3 +73,182 @@ func TestBuildDupTask(t *testing.T) {
+	data3IndexKey := kvPairs.Pairs[5].Key
+	data3IndexValue := kvPairs.Pairs[5].Val
+
+	ctx := context.Background()
+
+	originalErr := common.ErrFoundDuplicateKeys.FastGenByArgs(data2RowKey, data2RowValue)
+
+	newErr := local.ConvertToErrFoundConflictRecords(ctx, originalErr, tbl)
+	require.EqualError(t, newErr, "[Lightning:Restore:ErrFoundDataConflictRecords]found data conflict records in table a, primary key is '2', row data is '[KindInt64 2 KindInt64 6 KindString 2.csv]'")
+
+	originalErr = common.ErrFoundDuplicateKeys.FastGenByArgs(data3IndexKey, data3IndexValue)
+
+	newErr = local.ConvertToErrFoundConflictRecords(ctx, originalErr, tbl)
+	require.EqualError(t, newErr, "[Lightning:Restore:ErrFoundIndexConflictRecords]found index conflict records in table a, unique key is '[7]', primary key is '3'")
+}
+
+func TestConvertToErrFoundConflictRecordsMultipleColumnsIndex(t *testing.T) {
+	p := parser.New()
+	node, _, err := p.ParseSQL("create table a (a int primary key, b int not null, c text, d int, unique key key_bd(b,d));")
+	require.NoError(t, err)
+	mockSctx := mock.NewContext()
+	info, err := ddl.MockTableInfo(mockSctx, node[0].(*ast.CreateTableStmt), 108)
+	require.NoError(t, err)
+	info.State = model.StatePublic
+	tbl, err := tables.TableFromMeta(lkv.NewPanickingAllocators(0), info)
+	require.NoError(t, err)
+
+	sessionOpts := encode.SessionOptions{
+		SQLMode:   mysql.ModeStrictAllTables,
+		Timestamp: 1234567890,
+	}
+
+	encoder, err := lkv.NewBaseKVEncoder(&encode.EncodingConfig{
+		Table:          tbl,
+		SessionOptions: sessionOpts,
+		Logger:         log.L(),
+	})
+	require.NoError(t, err)
+	encoder.SessionCtx.GetSessionVars().RowEncoder.Enable = true
+
+	data1 := []types.Datum{
+		types.NewIntDatum(1),
+		types.NewIntDatum(6),
+		types.NewStringDatum("1.csv"),
+		types.NewIntDatum(101),
+	}
+	data2 := []types.Datum{
+		types.NewIntDatum(2),
+		types.NewIntDatum(6),
+		types.NewStringDatum("2.csv"),
+		types.NewIntDatum(102),
+	}
+	data3 := []types.Datum{
+		types.NewIntDatum(3),
+		types.NewIntDatum(7),
+		types.NewStringDatum("3.csv"),
+		types.NewIntDatum(103),
+	}
+	tctx := encoder.SessionCtx.GetTableCtx()
+	_, err = encoder.Table.AddRecord(tctx, data1)
+	require.NoError(t, err)
+	_, err = encoder.Table.AddRecord(tctx, data2)
+	require.NoError(t, err)
+	_, err = encoder.Table.AddRecord(tctx, data3)
+	require.NoError(t, err)
+	kvPairs := encoder.SessionCtx.TakeKvPairs()
+
+	data2RowKey := kvPairs.Pairs[2].Key
+	data2RowValue := kvPairs.Pairs[2].Val
+	data3IndexKey := kvPairs.Pairs[5].Key
+	data3IndexValue := kvPairs.Pairs[5].Val
+
+	ctx := context.Background()
+
+	originalErr := common.ErrFoundDuplicateKeys.FastGenByArgs(data2RowKey, data2RowValue)
+
+	newErr := local.ConvertToErrFoundConflictRecords(ctx, originalErr, tbl)
+	require.EqualError(t, newErr, "[Lightning:Restore:ErrFoundDataConflictRecords]found data conflict records in table a, primary key is '2', row data is '[KindInt64 2 KindInt64 6 KindString 2.csv KindInt64 102]'")
```

Comment:
`[KindInt64 2 KindInt64 6 KindString 2.csv KindInt64 102]` is not very good in reading, you can use `decoder.DecodeRawRowDataAsStr` to get a more friendly string